Add some documentation for podman (#262)
mila-iqia/mila-docs: docs/Userguide_containers.rst

Using containers
================

> Docker containers are now available on the local cluster without root
> priviledges using `podman <https://podman.io>`_.

Reviewer: ```suggestion
Podman containers are now available as tech preview on the Mila cluster
without root privileges using `podman <https://podman.io>`_.
```

a local folder inside the container, such as `$SCRATCH` or your home to save
data.

.. code-block:: bash

   $ podman run --mount type=bind,source=$SCRATCH/exp,destination=/data/exp bash touch /data/exp/file
   $ ls $SCRATCH/exp
   file

You can use multiple containers in a single job, but you have to be careful
about the memory and CPU limits of the job.

.. note::

   Due to the cluster environment you may see warning messages like
   `WARN[0000] "/" is not a shared mount, this could cause issues or missing mounts with rootless containers`,
   `ERRO[0000] cannot find UID/GID for user <user>: no subuid ranges found for user "<user>" in /etc/subuid - check rootless mode in man pages.`,
   `WARN[0000] Using rootless single mapping into the namespace. This might break some images. Check /etc/subuid and /etc/subgid for adding sub*ids if not using a network user`
   or
   `WARN[0005] Failed to add pause process to systemd sandbox cgroup: dbus: couldn't determine address of session bus`
   but as far as we can see those can be safely ignored and should not have
   an impact on your images.

GPU
---

> To use a GPU in a container, you need to a GPU job and then use ``--device

Reviewer: You need to "queue" a GPU job ?

Using containers
================

> Docker containers are now available on the local cluster without root priviledges using `podman <https://podman.io>`_.

Reviewer 2: ```suggestion
Docker containers are now available on the local cluster without root
priviledges using `podman <https://podman.io>`_.
```

> Generally any command-line argument accepted by docker will work with podman. this means that you can mostly use docker examples in you find on the web by replacing docker with podman in the command line.

Reviewer 2: ```suggestion
Generally any command-line argument accepted by docker will work with podman.
This means that you can mostly use the docker examples you find on the web by
replacing `docker` with `podman` in the command line.
```

> .. note::
>     Complete Podman Documentation:
>     https://docs.podman.io/en/stable/

Reviewer 2: ```suggestion
.. note::
    Complete Podman Documentation: https://docs.podman.io/en/stable/
```

Using in SLURM
--------------

> To use podman you can just use the podman command in either a batch script or an interactive job.

Reviewer 2: ```suggestion
To use podman you can just use the `podman` command in either a batch script or
an interactive job.
```

> One difference in configuration is that for certain technical reasons all the storage for podman (images, containers, ...) is on a job-specific location and will be lost after the job is complete or preempted. If you have data that must be preseved across jobs, you can `mount <https://docs.podman.io/en/v5.2.4/markdown/podman-run.1.html#mount-type-type-type-specific-option>`_ a local folder inside the container, such as $SCRATCH or you home to save data.

Reviewer 2: ```suggestion
One difference in configuration is that for certain technical reasons all the
storage for podman (images, containers, ...) is on a job-specific location and
will be lost after the job is complete or preempted. If you have data that must
be preseved across jobs, you can `mount
<https://docs.podman.io/en/v5.2.4/markdown/podman-run.1.html#mount-type-type-type-specific-option>`_
a local folder inside the container, such as `$SCRATCH` or your home to save
data.
```

.. code-block:: bash

   $ podman run --mount type=bind,source=$SCRATCH/exp,destination=/data/exp bash touch /data/exp/file
   $ ls $SCRATCH/exp
   file

> You can use multiple containers in a single job, but you have to be careful about the memory and CPU limits of the job.

Reviewer 2: ```suggestion
You can use multiple containers in a single job, but you have to be careful
about the memory and CPU limits of the job.
```

GPU
---

> To use a GPU you need to a GPU job and then use the ``--device nvidia.com/gpu=all`` for all GPUs allocated to the job or ``--device nvidia.com/gpu=n`` where n is the gpu you want in the container, starting at 0.

Reviewer 2: ```suggestion
To use a GPU in a container, you need to a GPU job and then use ``--device
nvidia.com/gpu=all`` to make all GPUs allocated available in the container or
``--device nvidia.com/gpu=N`` where `N` is the gpu index you want in the
container, starting at 0.
```
